packages/components/src/popover/index.js

```diff
@@ -602,25 +603,15 @@ const Popover = ( {
 		content = <FocusManaged>{ content }</FocusManaged>;
 	}
 
-	return (
-		<Consumer>
-			{ ( { getSlot } ) => {
-				// In case there is no slot context in which to render,
-				// default to an in-place rendering.
-				if ( getSlot && getSlot( __unstableSlotName ) ) {
-					content = (
-						<Fill name={ __unstableSlotName }>{ content }</Fill>
-					);
-				}
+	if ( slot.ref ) {
+		content = <Fill name={ __unstableSlotName }>{ content }</Fill>;
+	}
 
-				if ( anchorRef || anchorRect ) {
-					return content;
-				}
+	if ( anchorRef || anchorRect ) {
+		return content;
+	}
 
-				return <span ref={ anchorRefFallback }>{ content }</span>;
-			} }
-		</Consumer>
-	);
+	return <span ref={ anchorRefFallback }>{ content }</span>;
```

Comment:
[Looks like PopoverSlot always uses bubblesVirtually](https://github.com/WordPress/gutenberg/blob/0b11256824a1aae503ade5d7071c42d2b2c354ce/packages/components/src/popover/index.js#L619-L621), so I also changed this snipper to use `useSlot` instead of `Consumer`. With that, `Consumer` is only used on the `SlotFill` module internally.